backend/app/modules/access/auth_controller.py

```diff
@@ -301,3 +303,177 @@ def confirm_forgot_password(body: ConfirmForgotPasswordRequest,
                             })
 
     return {"message": "Password reset successful"}
+
+
+
+
+@router.post("/invite",
+             description="Invites a new user to application after creating a new account with user email and a temporary password in AWS Cognito.",
+             response_model=InviteResponse)
+def invite(body: InviteRequest, 
+           request: Request, 
+           settings: SettingsDep,
+           db: DbSessionDep,
+           cognito_client: CognitoIdpDep):
+    
+    id_token = request.cookies.get('id_token')
+    refresh_token = request.cookies.get('refresh_token')
+
+    if None in (refresh_token, id_token):
+        raise HTTPException(status_code=401,
+                            detail="Missing refresh token or id token")
+
+    decoded_id_token = jwt.decode(id_token,
+                                  algorithms=["RS256"],
+                                  options={"verify_signature": False})
+
+    coordinator_email = decoded_id_token.get('email')
+    if not coordinator_email:
+        raise HTTPException(status_code=401,
+                            detail="Missing 'email' field in the decoded ID token.")
+
+    numbers = '0123456789'
+    lowercase_chars = 'abcdefghijklmnopqrstuvwxyz'
+    uppercase_chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
+    symbols = '.-_~'
+    temporary_password = ''.join(random.choices(numbers, k=3)) + ''.join(random.choices(lowercase_chars, k=3)) + ''.join(random.choices(symbols, k=1)) + ''.join(random.choices(uppercase_chars, k=3))
+
+    try:
+        cognito_client.admin_create_user(
+            UserPoolId=settings.COGNITO_USER_POOL_ID,
+            Username=request.email,
+            TemporaryPassword=temporary_password,
+            ClientMetadata={
+                'url': settings.ROOT_URL
+            },
+            DesiredDeliveryMediums=["EMAIL"]
+        )
+
+    except ClientError as error:
+        if error.response['Error']['Code'] == 'UserNotFoundException':
+            raise HTTPException(status_code=400, detail="User not found. Confirmation not sent.")
+        else:
+            raise HTTPException(status_code=500, detail=error.response['Error']['Message'])
+   
+    try:
+      
+        user = create_user(db, UserCreate(
```

Comment:
In later development, we should update the CRUD functions to accept SQLAlchemy models instead of pydantic data schemas. This will allow clearer separation of tech and establish clearer responsibilities.